.github/workflows/build-multiarch.yaml

```
@@ -0,0 +1,323 @@
+            org.opencontainers.image.description=Builder Image for GOV.UK Ruby-based Apps
+            org.opencontainers.image.source=https://github.com/alphagov/govuk-ruby-images
+            org.opencontainers.image.version=${{ join(matrix.version.rubyver, '.') }}
+            org.opencontainers.image.created=${{ steps.calculate-image-tags.outputs.createdDate }}
+            org.opencontainers.image.vendor=GDS
+          tags: |
+            type=semver,pattern={{raw}},suffix=-arm64,value=${{ join(matrix.version.rubyver, '.') }}
+            type=raw,value=latest-arm64,enable=${{ matrix.version.extra == 'latest' }}
+            type=sha,enable=true,prefix=${{ join(matrix.version.rubyver, '.') }}-,suffix=-arm64,format=short
+            type=sha,enable=true,priority=100,prefix=${{ join(matrix.version.rubyver, '.') }}-,suffix=-arm64,format=long
+
+      - name: Build Base Image
+        id: build-base-image
+        uses: docker/build-push-action@v5
+        with:
+          file: base.Dockerfile
+          context: .
+          push: ${{ inputs.pushToRegistry || true }}
+          platforms: "linux/arm64"
+          provenance: false
+          build-args: |
+            RUBY_MAJOR=${{ matrix.version.rubyver[0] }}.${{ matrix.version.rubyver[1] }}
+            RUBY_VERSION=${{ join(matrix.version.rubyver, '.') }}
+          tags: ${{ steps.base-image-metadata.outputs.tags }}
+          labels: ${{ steps.base-image-metadata.outputs.labels }}
+
+      - name: Build Builder Image
+        id: build-builder-image
+        uses: docker/build-push-action@v5
+        with:
+          file: base.Dockerfile
+          context: .
+          push: ${{ inputs.pushToRegistry || true }}
+          platforms:  "linux/arm64"
+          provenance: false
+          build-args: |
+            RUBY_MAJOR=${{ matrix.version.rubyver[0] }}.${{ matrix.version.rubyver[1] }}
+            RUBY_VERSION=${{ join(matrix.version.rubyver, '.') }}
+            OWNER=${{ github.repository_owner }}
+          tags: ${{ steps.builder-image-metadata.outputs.tags }}
+          labels: ${{ steps.builder-image-metadata.outputs.labels }}
+
+  create_docker_manifests:
+    if: ${{ inputs.pushToRegistry || true }}
+    name: Create Docker Manifests for Ruby ${{ join(matrix.version.rubyver, '.') }} Images
+    needs:
+      - build_and_push_amd64
+      - build_and_push_arm64
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        version: [
+          {rubyver: ['3', '1', '4'], extra: ''},
+          {rubyver: ['3', '2', '2'], extra: ''},
+          {rubyver: ['3', '3', '0'], extra: 'latest'},
+        ]
+    permissions:
+      packages: write
+    steps:
+      - name: Login to GHCR
+        uses: docker/login-action@v3.0.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout from GitHub
+        uses: actions/checkout@v4.1.1
```

No need to to pin GitHub's built-in actions to patch versions (toil reduction). Same throughout and for the Docker ones.
```suggestion
        uses: actions/checkout@v4
```